src/operators/OperatorStatistics.h

```diff
@@ -32,14 +32,14 @@
 
 namespace mrcpp {
 
-template <int D> class OperatorStatistics final {
+  template <int D, typename T> class OperatorStatistics final {
 public:
     OperatorStatistics();
     ~OperatorStatistics();
 
     void flushNodeCounters();
-    void incrementFNodeCounters(const MWNode<D> &fNode, int ft, int gt);
-    void incrementGNodeCounters(const MWNode<D> &gNode);
+    void incrementFNodeCounters(const MWNode<D, T> &fNode, int ft, int gt);
```

Comment:
Actually, I don't think this class should be templated on both `D` and `T` at all. It might be enough to simply template the methods that accept a `MWNode` object :thinking: